pkg/process/procutil/data_scrubber_windows.go

```diff
@@ -20,3 +25,66 @@ var (
 	// it's for handling parameters like `/p` and `/rp` in windows
 	forbiddenSymbolsRegex = "[^/a-zA-Z0-9_*]"
 )
+
+var executibleExtensions = []string{".com", ".exe", ".bat", ".cmd", ".vbs", ".vbe", ".js", ".jse", ".wsf", ".wsh", ".psc1", ".ps1"}
+
+// stripArguments identifies windows extension and extracts command. Otherwise, returns the first element of the cmdline before first space.
+// If cmdline is empty, stripArguments will return an empty string.
+func (ds *DataScrubber) stripArguments(cmdline []string) []string {
+	if len(cmdline) < 1 {
+		return cmdline
+	}
+
+	strCmdline := cmdline[0]
+
+	// Case 1: OS has already completed splitting as there is one token per element, we return first token.
+	if len(cmdline) > 1 {
+		return []string{strCmdline}
+	}
+
+	// Case 2: One string for cmdline, use extensionParser() to find first token.
+	if len(cmdline) == 1 && !strings.HasPrefix(strCmdline, "\"") {
+		strippedCmdline := extensionParser(strCmdline, executibleExtensions)
+
+		if strippedCmdline != "" {
+			return []string{strings.TrimSuffix(strippedCmdline, " ")}
+		}
+
+		// If no extension is found, return first token of cmdline.
+		return []string{strings.TrimSuffix(strings.SplitN(strCmdline, " ", 2)[0], " ")}
```

Comment:
Is the `TrimSuffix` call necessary? `SplitN` would already guarantee there are no spaces at the end of the substring, right?

Reply:
Right - will remove this in the next commit!